src/Core/src/Platform/iOS/ImageViewExtensions.cs

```diff
@@ -21,24 +22,41 @@ public static void UpdateAspect(this UIImageView imageView, IImage image)
 			imageView.ClipsToBounds = imageView.ContentMode == UIViewContentMode.ScaleAspectFill || imageView.ContentMode == UIViewContentMode.Center;
 		}
 
-		public static void UpdateIsAnimationPlaying(this UIImageView imageView, IImageSourcePart image)
+		public static async Task UpdateIsAnimationPlaying(this UIImageView imageView, IImageSourcePart image)
 		{
-			if (image.IsAnimationPlaying)
+			if (imageView is MauiImageView mauiImageView)
 			{
-				if (!imageView.IsAnimating)
-					imageView.StartAnimating();
-			}
-			else
-			{
-				if (imageView.IsAnimating)
-					imageView.StopAnimating();
+				// IsAnimationPlaying set to true in the incoming imagesource
+				// indicates it's an animated image.
+				if (image.IsAnimationPlaying)
+				{
+					if (mauiImageView.IsAnimating)
+					{
+						mauiImageView.StopAnimating();
+					}
+					
+					if (!imageView.IsAnimating)
+					{
+						var animatedImage = await ImageAnimationHelper.CreateAnimationFromImageSource(image.Source);
```

Comment:
Why do we need to load here? Should not setting the source do a check after loading to make sure the animation is running? The toggle can probably set animation on/off and then when the image is loaded, the check can be done to start the animation.

Reply:
Maybe this loading logic should be in the image source services. Then when we use the images in other places, GIF still works and is not restricted to _just_ image views. And then the coad can be the same since the only change is in the services.

Reply:
I also see starting and stopping re-downloads, re-parses and re-builds the image.

Reply:
Yeah I guess that makes sense (also for the comment below), but maybe that's something for another time? For Android (and Windows?) it currently works like this too. I'd think that if we want to fix this like that we should do it for _all_ platforms?

Reply:
In Xamarin.Forms the image was loaded in the `IImageSourceHandler`  impl https://github.com/xamarin/Xamarin.Forms/blob/88b70ad2824a4f40fa2ef056439ada9dc3e85ab4/Xamarin.Forms.Platform.iOS/Renderers/ImageRenderer.cs#L249 Could be possible to do the same?